src/main/java/com/exasol/dbbuilder/dialects/postgres/PostgreSqlUser.java

```diff
@@ -0,0 +1,47 @@
+package com.exasol.dbbuilder.dialects.postgres;
+
+import com.exasol.db.Identifier;
+import com.exasol.dbbuilder.dialects.*;
+
+/**
+ * PostgreSQL {@link User}.
+ */
+public class PostgreSqlUser extends AbstractUser {
+    private final PostgreSqlImmediateDatabaseObjectWriter writer;
+
+    /**
+     * Create a PostgreSQL user.
+     * 
+     * @param writer object writer
+     * @param name   username
+     */
+    public PostgreSqlUser(final PostgreSqlImmediateDatabaseObjectWriter writer, final Identifier name) {
+        super(name);
+        this.writer = writer;
+        writer.write(this);
```

Comment:
I find this dangerous, as we are calling some behavior on an object that is not yet fully initialized.